andela/ah-infinity-stones
# #161966918 User Comments

#### What does this PR do?
- Integrates user ability to leave comments on articles and threads
- Allows comment owner to edit or delete comments

#### Description of Task to be completed?
> User navigates to provided URLs or endpoints in the browser
> User should access the live application

#### How should this be manually tested?
`$ git clone -b ft-user-comments-161966918 https://github.com/andela/ah-infinity-stones`
`$ cd ah-infinity-stones`
> Create and activate a virtual envirinment to hold this app's dependancies
`$ virtualenv -p python3 venv`
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install dependencies`
> Setup environment variables in a file called `.env` with your own values. See sample below
> Load the `.env` variables by `$ source .env`
> Prepare migrations: `$ python3 manage.py makemigrations` and
> Run migrations: `$ python3 manage.py migrate`
> Run generic tests `$ python3 manage.py test`
> Run specific tests `$ python3 manage.py test authors/apps/comments`
> Register for a token to use on protected endpoint unless Read Only
> Engage the following endpoints:

> `POST http://127.0.0.1:8000/api/users/` to register
> `POST http://127.0.0.1:8000/api/articles/<slug>/comments`
> `GET http://127.0.0.1:8000/api/articles/<slug>/`
> `GET PUT DELETE http://127.0.0.1:8000/api/articles/<slug>/comments/1`

Populate your `.env` with:
```
export DATABASE_URL=postgres://<database_user>:<database_password>@localhost/<database_name>

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 export SECRET_KEY='your_secret'
```

#### What are the relevant pivotal tracker stories?
[#161966918](https://www.pivotaltracker.com/story/show/161966918)

#### Screenshots
<img width="1280" alt="screen shot 2019-01-08 at 10 05 49" src="https://user-images.githubusercontent.com/39657549/50815310-b43f1580-132d-11e9-80b0-49ba306c5e22.png">

<img width="1280" alt="screen shot 2019-01-08 at 10 08 23" src="https://user-images.githubusercontent.com/39657549/50815312-b43f1580-132d-11e9-8b5f-0af2202ce184.png">

<img width="1280" alt="screen shot 2019-01-08 at 10 09 14" src="https://user-images.githubusercontent.com/39657549/50815313-b4d7ac00-132d-11e9-9644-162087ec400d.png">

<img width="1280" alt="screen shot 2019-01-08 at 10 09 23" src="https://user-images.githubusercontent.com/39657549/50815315-b4d7ac00-132d-11e9-908b-00f5ed7787bb.png">
